spec/rubocop/cop/style/redundant_regexp_escape_spec.rb

```diff
@@ -533,4 +533,17 @@
       RUBY
     end
   end
+
+  context 'with escaping invalid byte sequence in UTF-8' do
+    it 'registers an offense and corrects' do
+      expect_offense(<<~'RUBY')
+        r = /[\§]/
```

Comment:
JRuby CI matrices are failing. Can you investigate? You can also check locally.

```console
% rbenv local jruby-9.3.6.0
% bundle exec rspec spec/rubocop/cop/style/redundant_regexp_escape_spec.rb:537
Run options: include {:locations=>{"./spec/rubocop/cop/style/redundant_regexp_escape_spec.rb"=>[537]}}

Randomized with seed 51753
F

Failures:

  1) RuboCop::Cop::Style::RedundantRegexpEscape with escaping invalid byte sequence in UTF-8 registers an offense and corrects
     Failure/Error:
       raise 'Error parsing example code: ' \
             "#{processed_source.diagnostics.map(&:render).join("\n")}"

     RuntimeError:
       Error parsing example code: (string):1:5: error: invalid multibyte character: /[\Â§]/
       (string):1: r = /[\§]/
       (string):1:     ^~~~~~
     # ./lib/rubocop/rspec/expect_offense.rb:201:in `parse_processed_source'
     # ./lib/rubocop/rspec/expect_offense.rb:119:in `expect_offense'
     # ./spec/rubocop/cop/style/redundant_regexp_escape_spec.rb:539:in `block in <main>'

Finished in 0.73504 seconds (files took 4.91 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/rubocop/cop/style/redundant_regexp_escape_spec.rb:538 # RuboCop::Cop::Style::RedundantRegexpEscape with escaping invalid byte sequence in UTF-8 registers an offense and corrects
```

Reply:
I am reporting an Issue because the code for the test case I added this time in JRuby seems to have a syntax error.
- https://togithub.com/jruby/jruby/issues/7306

Reply:
A corrected version has been released; CI is running again.
- https://togithub.com/jruby/jruby/pull/7314

Reply:
Jruby 9.3.7.0 has been released. Rerun the CI.

Reply:
https://hub.docker.com/r/circleci/jruby hasn't been updated in a while.
After the following PR merge we will move forward.

- https://github.com/rubocop/rubocop/pull/10932